cmake/SearchForStuff.cmake

```diff
@@ -326,6 +326,22 @@ if (PKG_CONFIG_FOUND)
     endif (NOT TBB_FOUND)
   endif (NOT TBB_FOUND)
 
+  # tbb::task was removed in tbb v2021.01, so in this case we must move to the
+  # newer tbb::task_group API. This could also be used with earlier versions of
+  # TBB, but we want to maintain ABI compatibility for Ubuntu 18.04 etc., so
+  # disable this by default.
+  if (NOT DEFINED USE_LEGACY_TBB_TASK)
+    if (${TBB_VERSION} VERSION_LESS 2021)
+      set (USE_LEGACY_TBB_TASK ON)
+    else (${TBB_VERSION} VERSION_LESS 2021)
+      set (USE_LEGACY_TBB_TASK OFF)
+    endif(${TBB_VERSION} VERSION_LESS 2021)
+  endif()
+  set(USE_LEGACY_TBB_TASK ${USE_LEGACY_TBB_TASK} CACHE BOOL "Whether to use the deprecated tbb::task class")
```

Comment:
Nitpick, feel free to ignore: in my experience cache variable vs normal variables is one aspect of CMake that is most confusing for novice users. To improve readability, it may be worth to use a different name for the normal variable name, like `USE_LEGACY_TASK_DEFAULT_VALUE`.